plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/reporting/FormattedError.kt

```diff
@@ -50,7 +50,10 @@ class CallsInPlaceError(private val sourceRole: SourceRole.CallsInPlaceEffect) :
 
 class LeakingLambdaError(private val error: VerificationError) : FormattedError {
     override fun report(reporter: DiagnosticReporter, source: KtSourceElement?, context: CheckerContext) {
-        reporter.reportOn(source, PluginErrors.LAMBDA_MAY_LEAK, error.reason.fetchLeakingFunction(), context)
+        // The leaking function symbol is always contained in the first argument of the error's reason.
+        val reasonArg = error.reason.asCallable().argInfo(0)
```

Comment:
This setup seems to conflate `error` and `reason` with their offending node.  Maybe this makes sense for `reason`, but in that case why not get rid of the `ErrorReason` class entirely; just make a node representation and have `error` contain two `nodes` for its own offending node and for the reason's node.  (It isn't quite clear to me how Viper decides what to put in those two; perhaps that would be good to document somewhere.)

In general, it seems like a lot of the effort here goes to avoiding explicitly talking about nodes.  Is explicitly representing nodes not an option?

Reply:
> In general, it seems like a lot of the effort here goes to avoiding explicitly talking about nodes. Is explicitly representing nodes not an option?

Yeah, sadly we cannot expose the Viper's `Node` class explicitly (due to package conflicts, such as: `Cannot access 'scala.Equals' which is a supertype of 'viper.silver.ast.Node'. Check your module classpath for missing or conflicting dependencies`).

Reply:
But you can expose it via the same kind of wrapper as you've used for all these others, can't you?

Reply:
Yes, I can. But, we are going to have a single class for Node:

```kotlin
class Node(val node: viper.ast.silver.Node) 
```

And then I can extend it with the `IntoSilverCallable`. WDYT?

Reply:
I don't see any problem